### PR TITLE
Update potion effect properties to modern format

### PR DESCRIPTION
**Pull request by tal5**
## Changes
To `EntityPotionEffects` and `ItemPotion`:

- Added `effects_data` tags which return a `ListTag(MapTag)` (The modern format)
- Added `MapTag/ListTag(MapTag)` input support to potion effect mechanisms

## Technical

- Removed `getPotionEffect` from `ItemHelper`, as it's the same on all supported versions
- Added `get/isArrow` methods to `ItemPotion`
- Added `getEffectsList/MapTag` methods to `EntityPotionEffects`, due to duplicate code
- Changed `EntityTag.has_effect` to use `LivingEntity#hasPotionEffect` / `Arrow#hasCustomEffect` instead of looping over all existing effects
- Added a `ItemPotion#parseEffect` method for MapTags
- Added `ItemPotion#effectToMap`
- Changed `ItemPotion#describes` to check for the Item's meta being an `instanceof PotionMeta`
- Added `ItemPotion#getMapTagData` (Used for `ItemTag.effects_data` and `ItemPotion#getPropertyString`)

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPotionEffects.java [hunk 1]
```diff
@@ -52,33 +50,46 @@ private EntityPotionEffects(EntityTag entity) {
...
+    public boolean isArrow() {
+        return entity.getBukkitEntity() instanceof Arrow;
+    }
+
+    public Arrow getArrow() {
+        return (Arrow) entity.getBukkitEntity();
+    }
+
+    public String getPropertyString() {
+        ListTag effects = getEffectsListTag();
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPotionEffects.java]: Should probably use the map version here?

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPotionEffects.java [hunk 2]
```diff
@@ -136,28 +158,40 @@ public void adjust(Mechanism mechanism) {
...
         // @tags
+        // <EntityTag.effects_data>
         // <EntityTag.list_effects>
         // <EntityTag.has_effect[<effect>]>
         // -->
         if (mechanism.matches("potion_effects")) {
-            ListTag effects = ListTag.valueOf(mechanism.getValue().asString(), mechanism.context);
-            for (String effectStr : effects) {
-                PotionEffect effect = ItemPotion.parseEffect(effectStr, mechanism.context);
+            Collection<ObjectTag> effects = CoreUtilities.objectToList(mechanism.value, mechanism.context);
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPotionEffects.java]: variable is redundant, value should just be used directly in the `for` line

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/item/ItemPotion.java [hunk 3]
```diff
@@ -93,15 +101,87 @@ public static PotionEffect parseEffect(String str, TagContext context) {
...
+
+    public PotionMeta getMeta() {
+        return (PotionMeta) item.getItemMeta();
+    }
+
+    @Override
+    public String getPropertyString() {
+        PotionMeta meta = getMeta();
         ListTag effects = new ListTag();
         effects.add(meta.getBasePotionData().getType()
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/item/ItemPotion.java]: Should update to the map return here as well

**Comment by tal5** [comment]
All should be fixed

**Pull request merged**